test/functional/tests/data_integrity/test_data_integrity_with_patterns.py

```diff
@@ -0,0 +1,259 @@
+        if fio_errors:
+            TestRun.LOGGER.error(f"Found {fio_errors} errors in fio output")
+
+    with TestRun.step("Delete test files."):
+        for file in fio_targets:
+            remove(file, True, True, True)
+
+    with TestRun.step("Unmount all cores."):
+        for core in cores:
+            core.unmount()
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_data_integrity_mixed_patterns(cache_mode):
+    """
+        title:
+          Data integrity test with mixed write patterns with duration time 2 hours
+        description: |
+          Run fio on each core with different pattern and verify if there's no write errors.
+        pass_criteria:
+            - System does not crash.
+            - Fio does not return any errors.
+    """
+    with TestRun.step("Prepare cache and core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([exp_obj_size * exp_obj_number])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([exp_obj_size] * exp_obj_number)
+
+    with TestRun.step(f"start cache in {cache_mode} mode"):
+        cache = casadm.start_cache(cache_part, cache_mode, force=True)
+
+    with TestRun.step(f"Add all core devices to cache."):
+        cores = []
+        for core_part in core_dev.partitions:
+            core_part.create_filesystem(Filesystem.ext3, True)
+            cores.append(cache.add_core(core_part))
+
+    with TestRun.step("Mount all cores"):
+        fio_targets = []
+        for i, core in enumerate(cores):
+            core_mount_point = f"{mount_point}{core.core_id}"
+            core.mount(f"{core_mount_point}")
+            fio_targets.append(f"{core_mount_point}/test_file")
+
+    with TestRun.step(f"Select {exp_obj_number} different patterns, one for each core."):
+        patterns = mix_patterns()
+
+    with TestRun.step(f"Run fio with mixed pattern."):
+        fio = prepare_mixed_fio(fio_targets, patterns)
+        fio.run()
+
+    with TestRun.step("Check for fio errors."):
+        fio_output = TestRun.executor.run(f"cat {fio.fio.fio_file}")
+        fio_errors = fio.get_results(fio_output.stdout)[0].total_errors()
+        if fio_errors:
+            TestRun.LOGGER.error(f"Found {fio_errors} errors in fio output")
+
+    with TestRun.step("Delete test files."):
+        for file in fio_targets:
+            remove(file, True, True, True)
+
+    with TestRun.step("Unmount all cores."):
+        for core in cores:
+            core.unmount()
+
+    with TestRun.step("Stop cache."):
+        cache.stop()
+
+
+def prepare_unified_fio(targets, data_pattern):
+    fio = _prepare_fio()
+    fio.verification_with_pattern(data_pattern) \
+        .set_param("do_verify", 0)
```

Comment:
When you turn off verification, you're ridding the test of its core function - data integrity verification. In this test we don't focus on io errors but rather on verification errors.